### ISSUE TITLE
Incorrect nullability warning in assignment to generic extension property

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#81824]
There should be no warning in the following example:
```csharp
    [Fact]
    public void Nullability_PropertyAccess_07()
    {
        var src = """
#nullable enable

object? oNotNull = new object();
oNotNull.P = null; // 1
E.set_P(oNotNull, null);

static class E
{
    extension<T>(T t)
    {
        public T P { set => throw null!; }
    }
}
""";
        var comp = CreateCompilation(src);
        comp.VerifyEmitDiagnostics(
            // (4,14): warning CS8625: Cannot convert null literal to non-nullable reference type.
            // oNotNull.P = null; // 1
            Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 14));
    }
```

Relates to test plan https://github.com/dotnet/roslyn/issues/76130

**Comment by lakshay122007** [comment]
Hi!
i have raised a PR for this issue which resolves it. Please kindly review it and if any further changes are required please let me know, i will fix them. And if everything is good, hope it gets merged!
